analysis/04_correlation_with_expression/TE_density_plot_expression.Rmd

```diff
@@ -0,0 +1,203 @@
+                        isPairedEnd = TRUE,
+                        nthreads = 16)
+
+# Here we will find the number of reads per kilobases of a given gene body (rpkm).
+rpk <- counts$counts / (counts$annotation$Length/1000)
+expression <- data.frame("rpk" = rpk) %>%
+  rownames_to_column("gene_id") %>%
+  pivot_longer(2:3, names_to = "sample", values_to = "rpk")
+
+# Here we will convert rpkm to transcripts per million of reads (tpm)
+expression_summary <- expression %>%
+  group_by(sample) %>%
+  summarize(total_rpk = sum(rpk, na.rm = T))
+expression_summary$rpk_m <- expression_summary$total_rpk / 1e6
+expression <- merge(expression, expression_summary)
+expression$tpm <-expression$rpk / expression$rpk_m
+tpm <- expression %>% group_by(gene_id) %>%
+  summarize(tpm = mean(tpm, na.rm = T))
+```
+
+
+Obtain promoter regions for lncRNA and protein-coding genes.
+```{r}
+#Obtain necessary functions and plot themes
+source("../util/intersect_functions.R")
+source("../util/_setup.R")
+
+# Read in consensus peaks as a list.
+# import_peaks is a function
+peak_list <- import_peaks()
+
+# Read in gencode genome annotation.
+gtf <- rtracklayer::import("/Shares/rinn_class/data/genomes/human/gencode/v32/gencode.v32.annotation.gtf")
+
+# Get the promoter regions of each gene.
+# get_promoter_regions is a function
+proms_lncRNA <- get_promoter_regions(gtf, biotype = c("lncRNA"))
+proms_protein <- get_promoter_regions(gtf, biotype = c("protein_coding"))
+proms <- c(proms_lncRNA, proms_protein)
+
+# Count peaks over promoters
+# count_peaks_per_feature is a function
+promoter_peak_count <- count_peaks_per_feature(proms, peak_list)
+```
+
+
+```{r}
+# Make this binary -- a co-occurence matrix.
+promoter_peak_occurence <- matrix(as.numeric(promoter_peak_count > 0),
+                           nrow = dim(promoter_peak_count)[1],
+                           ncol = dim(promoter_peak_count)[2])
+rownames(promoter_peak_occurence) <- rownames(promoter_peak_count)
+colnames(promoter_peak_occurence) <- colnames(promoter_peak_count)
+
+#Pull gene IDs and transcript types out of gencode gtf
+gene_id <- gtf$gene_id
+transcript_type <- gtf$transcript_type
+
+#Make a data frame with gene IDs and transcript types.
+gene_type_df <- data.frame('gene_id' = gene_id, 'transcript_type' = transcript_type)
+gene_type_df <- gene_type_df[which(gene_type_df$transcript_type == c("lncRNA", "protein_coding")),]
+
+# Make a data frame of binding events per promoter.
+peak_occurence_df <- data.frame("gene_id" = colnames(promoter_peak_occurence),
+                                "number_of_tfs" = colSums(promoter_peak_occurence))
+
+# Merge in the expression data into the data frame you just created.
+peak_occurence_df <- merge(peak_occurence_df, gene_type_df)
+peak_occurence_df <- merge(peak_occurence_df, tpm)
+```
+
+
+```{r}
+# Filter your data frame to promoters that have more than x binding events yet are still not expressed.
+# Set x to 1
+x <- 1
```

Comment:
I think we've been using a cutoff of 7 TFs bound to define reservoirs, but for this plot I think a cutoff of one is fine.

Reply:
This reads really nice and delineates functions nicely -- if Michael approves I will merge